src/test/run-pass/issue-15149.rs

```diff
@@ -8,33 +8,44 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-use std::io::{TempDir, Command, fs};
+use std::path::BytesContainer;
+use std::io::{Command, fs, USER_RWX};
 use std::os;
 
 fn main() {
     // If we're the child, make sure we were invoked correctly
     let args = os::args();
     if args.len() > 1 && args[1].as_slice() == "child" {
-        return assert_eq!(args[0].as_slice(), "mytest");
+        return assert_eq!(args[0],
+                          format!("mytest{}", os::consts::EXE_SUFFIX));
     }
 
     test();
 }
 
 fn test() {
-    // If we're the parent, copy our own binary to a tempr directory, and then
-    // make it executable.
-    let dir = TempDir::new("mytest").unwrap();
-    let me = os::self_exe_name().unwrap();
-    let dest = dir.path().join(format!("mytest{}", os::consts::EXE_SUFFIX));
-    fs::copy(&me, &dest).unwrap();
-
-    // Append the temp directory to our own PATH.
+    // If we're the parent, copy our own binary to a new directory.
+    let my_path = os::self_exe_name().unwrap();
+    let my_dir  = my_path.dir_path();
+
+    let child_dir = Path::new(my_dir.join("issue-15149-child"));
```

Comment:
Could this add some random substring to the end of the directory as well? It's possible that this test is run a few times in parallel and it would help prevent conflicts